ci/scripts/install_pandas.sh

```diff
@@ -28,7 +28,7 @@ pandas=$1
 numpy=${2:-"latest"}
 
 if [ "${numpy}" = "nightly" ]; then
-  pip install --extra-index-url https://pypi.anaconda.org/scipy-wheels-nightly/simple --pre numpy
```

Comment:
scipy-wheels-nightly only has up to [pandas-2.1.0.dev0+926](https://pypi.anaconda.org/scipy-wheels-nightly/simple/pandas/2.1.0.dev0%2B926.g4e8306690e/pandas-2.1.0.dev0%2B926.g4e8306690e-cp39-cp39-manylinux_2_17_aarch64.manylinux2014_aarch64.whl)

scientific-python-nightly-wheels has up to
[pandas-2.2.0.dev0+325](https://pypi.anaconda.org/scientific-python-nightly-wheels/simple/pandas/2.2.0.dev0%2B325.g6c58a217f5/pandas-2.2.0.dev0%2B325.g6c58a217f5-cp39-cp39-macosx_11_0_arm64.whl)